### PR TITLE
Add a helper function for i2c clock setup

### PR DESCRIPTION
**Pull request by ChuckM**
The code lets you set up the i2c channel as slow (100Khz) or
fast (400Khz) taking care of setting up CR2/CCR/TRISE registers
for you. If you aren't doing interrupts or DMA you can start
sending data right after that.

### CODE REVIEW

File: lib/stm32/common/i2c_common_all.c [hunk 1]
```diff
@@ -458,4 +458,42 @@ void i2c_clear_dma_last_transfer(uint32_t i2c)
...
+
+	/* force disable, to set clocks */
+	I2C_CR1(i2c) &= ~(I2C_CR1_PE);
+
+	/* frequency in megahertz */
+	freq = rcc_apb1_frequency / 1000000;
+	reg = (I2C_CR2(i2c) & ~(I2C_CR2_FREQ_MASK)) | (freq & I2C_CR2_FREQ_MASK);
+	I2C_CR2(i2c) = reg;
+
+	/* Note supports the two 'simple' versions not the 9:16 ratio version */
```
Comment (karlp) [inline lib/stm32/common/i2c_common_all.c]: That sort of note is better up in the doxygen area I would expect

**Comment by ChuckM** [comment]
Ok, I think both comments have been covered.

**Comment by brabo** [comment]
Tested on STM32F429i-disco, everything compiles fine and works perfectly :)

**Comment by ChuckM** [comment]
@karlp @esden any reason this can't be merged?

**Comment by karlp** [comment]
Just didn't get back to it after the comments were addressed, will move it back up my least :)

File: lib/stm32/common/i2c_common_all.c [hunk 2]
```diff
@@ -458,4 +458,46 @@ void i2c_clear_dma_last_transfer(uint32_t i2c)
...
+void i2c_set_speed(uint32_t i2c, uint8_t fast)
+{
+	int	freq;
+	uint32_t reg;
+
+	/* force disable, to set clocks */
+	I2C_CR1(i2c) &= ~(I2C_CR1_PE);
+
+	/* frequency in megahertz */
+	freq = rcc_apb1_frequency / 1000000;
```
Comment (karlp) [inline lib/stm32/common/i2c_common_all.c]: are all  i2c periphs on ap1?

**Comment by karlp** [comment]
This has finally landed witth my staging 2017.4 branch

**Pull request closed without merging**